### PR TITLE
DAT-2941 replace comparison with extended group functionality

### PR DESCRIPTION
**Pull request by idradm**

### CODE REVIEW

File: extensions/wikia/PortableInfobox/services/Parser/Nodes/NodeGroup.php [hunk 1]
```diff
@@ -2,14 +2,21 @@
...
+	const LAYOUT_ATTR_NAME = 'layout';
+	const SHOW_ATTR_NAME = 'show';
+	const DEFAULT_OPTION = 'default';
+	const SHOW_INCOMPLETE_OPTION = 'incomplete';
 
 	private $supportedGroupLayouts = [
-		'default',
+		self::DEFAULT_OPTION,
 		'horizontal'
 	];
```
Comment (dianafa) [inline extensions/wikia/PortableInfobox/services/Parser/Nodes/NodeGroup.php]: Why don't we move this string to const as well?

Comment (idradm) [inline extensions/wikia/PortableInfobox/services/Parser/Nodes/NodeGroup.php]: There was no need for that yet

Comment (dianafa) [inline extensions/wikia/PortableInfobox/services/Parser/Nodes/NodeGroup.php]: :( ugly

**Comment by dianafa** [comment]
+68 −522 wohoo! LGTM

File: extensions/wikia/PortableInfobox/services/Parser/Nodes/NodeGroup.php [hunk 2]
```diff
@@ -2,14 +2,21 @@
...
-	const DATA_LAYOUT_ATTR_NAME = 'layout';
-	const DEFAULT_TAG_NAME = 'default';
+	const LAYOUT_ATTR_NAME = 'layout';
+	const SHOW_ATTR_NAME = 'show';
+	const DEFAULT_OPTION = 'default';
+	const SHOW_INCOMPLETE_OPTION = 'incomplete';
 
 	private $supportedGroupLayouts = [
-		'default',
+		self::DEFAULT_OPTION,
```
Comment (SebastianMarzjan) [inline extensions/wikia/PortableInfobox/services/Parser/Nodes/NodeGroup.php]: Would making this LAYOUT_OPTION_DEFAULT and below LAYOUT_OPTION_HORIZONTAL increase readability of that part?

Comment (SebastianMarzjan) [inline extensions/wikia/PortableInfobox/services/Parser/Nodes/NodeGroup.php]: If we go this direction, changing SHOW_INCOMPLETE_OPTION to SHOW_OPTION_INCOMPLETE might be a good thing for consistency

File: extensions/wikia/PortableInfobox/services/Parser/Nodes/NodeGroup.php [hunk 3]
```diff
@@ -20,9 +27,18 @@ public function getData() {
...
+				},
+				$this->getChildNodes()
+			)
+			: $this->getRenderDataForChildren();
+
 		return [
 			'type' => $this->getType(),
-			'data' => [ 'value' => $this->getRenderDataForChildren(),
+			'data' => [ 'value' => $value,
 						'layout' => $this->getLayout() ],
```
Comment (SebastianMarzjan) [inline extensions/wikia/PortableInfobox/services/Parser/Nodes/NodeGroup.php]: nitpick; spacing is confusing here :)

File: extensions/wikia/PortableInfobox/services/Parser/Nodes/NodeGroup.php [hunk 4]
```diff
@@ -42,13 +58,21 @@ public function getSource() {
 		return $this->getSourceForChildren();
 	}
 
-	/**
-	 * @return string
-	 */
+	protected function showIncomplete() {
+		return strcmp( $this->getDisplay(), self::SHOW_INCOMPLETE_OPTION ) === 0;
```
Comment (SebastianMarzjan) [inline extensions/wikia/PortableInfobox/services/Parser/Nodes/NodeGroup.php]: An interesting approach, I may be missing why we use strcmp an not a direct string comparison through === - what's the rationale as we're interested only in (in)equality and not its direction?

File: extensions/wikia/PortableInfobox/services/Parser/Nodes/NodeGroup.php [hunk 5]
```diff
@@ -2,14 +2,21 @@
...
+	const SHOW_INCOMPLETE_OPTION = 'incomplete';
 
 	private $supportedGroupLayouts = [
-		'default',
+		self::DEFAULT_OPTION,
 		'horizontal'
 	];
 
+	private $supportedGroupDisplays = [
+		self::DEFAULT_OPTION,
```
Comment (SebastianMarzjan) [inline extensions/wikia/PortableInfobox/services/Parser/Nodes/NodeGroup.php]: I'm battling whether we should reuse DEFAULT_OPTION widely or have an explicit default for each attribute (GROUP_OPTION_DEFAULT, etc).
Thoughts?

File: extensions/wikia/PortableInfobox/tests/PortableInfoboxRenderServiceTest.php [hunk 6]
```diff
@@ -311,72 +311,6 @@ public function testRenderInfoboxDataProvider() {
 			[
 				'input' => [
 					[
-						'type' => 'comparison',
```
Comment (SebastianMarzjan) [inline extensions/wikia/PortableInfobox/tests/PortableInfoboxRenderServiceTest.php]: I think it's worth adding a case that handles our main use cases for group with show attribute to keep covering it

File: extensions/wikia/PortableInfobox/services/Parser/Nodes/NodeGroup.php (hunk 4 above)
Comment (SebastianMarzjan) [inline extensions/wikia/PortableInfobox/services/Parser/Nodes/NodeGroup.php]: Per offline discussion, please make sure that case sensitivity is handles as you want it to be handled

File: extensions/wikia/PortableInfobox/services/Parser/Nodes/NodeGroup.php (hunk 5 above)
Comment (idradm) [inline extensions/wikia/PortableInfobox/services/Parser/Nodes/NodeGroup.php]: Is there any need for that?

Comment (SebastianMarzjan) [inline extensions/wikia/PortableInfobox/services/Parser/Nodes/NodeGroup.php]: Unambiguity :)

File: extensions/wikia/PortableInfobox/tests/PortableInfoboxRenderServiceTest.php (hunk 6 above)
Comment (idradm) [inline extensions/wikia/PortableInfobox/tests/PortableInfoboxRenderServiceTest.php]: It's already covered, as show option will change the way getRenderData works

Comment (SebastianMarzjan) [inline extensions/wikia/PortableInfobox/tests/PortableInfoboxRenderServiceTest.php]: :+1:

**Comment by SebastianMarzjan** [comment]
LG, please consider updating the const naming afterwards for greppability ( :heart: this word )

**Comment by SebastianMarzjan** [comment]
Wrong merge

**Pull request closed without merging**